specs/Popup-AdditionalLayoutProperties-Spec.md

```diff
@@ -0,0 +1,218 @@
+
+```xml
+<!-- Part of the CommandBarFlyoutCommandBar's default template -->
+<Popup
+    x:Name="OverflowPopup"
+    PlacementTarget="{Binding ElementName=PrimaryItemsRoot}"
+    DesiredPlacement="Bottom">
+</Popup>
+```
+
+```csharp
+void OnApplyTemplate()
+{
+    m_overflowPopup = GetTemplateChild("OverflowPopup");
+    m_overflowPopup.PlacementChanged += OnOverflowPopupPlacementChanged;
+}
+
+void OnOverflowPopupPlacementChanged(object sender, object args)
+{
+    UpdateVisualState(useTransitions: false);
+}
+
+void UpdateVisualState(bool useTransitions)
+{
+    if (m_overflowPopup.ActualPlacement == PopupPlacementMode.Top)
+    {
+        VisualStateManager.GoToState(this, "ExpandedUp", useTransitions);
+    }
+    else
+    {
+        VisualStateManager.GoToState(this, "ExpandedDown", useTransitions);
+    }
+}
+```
+
+## Popup.PlacementTarget property
+
+Use this property to describe which element the `Popup` should be positioned relative to.
+Defaults to `null`.
+
+If this is `null`, then `DesiredPlacement` is ignored, `ActualPlacement` is always `None`, and
+`PlacementChanged` is never raised.  If the `Popup` is in the visual tree, `PlacementTarget` will override what its
+position would otherwise be set to by layout.  Setting `PlacementTarget` to an element under a different XAML root than
+`Popup.XamlRoot` is invalid and will throw an `ArgumentException`.
+
+_Spec note: this property is analogous to the
+[FlyoutBase.Target](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Primitives.FlyoutBase.Target)
+property, but `Popup.Target` looked confusing, so we added the `Placement` prefix._
+
+## Popup.DesiredPlacement property
+
+Use this property to describe how the you would ideally like the `Popup`
+positioned relative to `PlacementTarget`.  Defaults to `None`.
+
+If this is `None`, then `PlacementTarget` is ignored,
+`ActualPlacement` is always `None` and `PlacementChanged` is never raised. 
+If both `DesiredPlacement` and `PlacementTarget` are set and `HorizontalOffset` and/or `VerticalOffset`
+are also set, then the latter two properties will offset the `Popup` from where it would have been
+placed by `DesiredPlacement` and `PlacementTarget` alone.
+
+## Popup.ActualPlacement property
+
+Use this read-only property to determine where the popup was positioned.
+Will always be `None` if either `PlacementTarget` and `DesiredPlacement` are not set.
+
+## Popup.PlacementChanged event
+
+Raised whenever XAML sets the value of `ActualPlacement`,
+which allows apps to respond to where a `Popup` was placed.
+
+For example, use this to determine the visual state to go into, 
+based on whether a `Popup` is appearing above or below `PlacementTarget`.
+
+This event is raised before the screen is refreshed, meaning that any visual changes made
+in response to this event can be made before anything is drawn to the screen at the new position.
+Will never be raised if either `PlacementTarget` and `DesiredPlacement` are not set.
```

Comment:
I would expect ActualPlacement to only update when the popup is being displayed, since there is no actual placement if it hasn't actually been placed anywhere.  So my feeling is that it should update to be None only if the popup is then displayed on screen with PlacementTarget = null.  I don't think ActualPlacement should update when the popup isn't open, since that will cause apps to do unnecessary work.

Reply:
I like saying that ActualPlacement only has meaning if it's open, and not raising the event when the popup closes.